### ISSUE TITLE
API/BUG: Categorical.is_dtype_equal doesn't compare to Series

### ISSUE THREAD

**Issue by chris-b1** [pandas-dev/pandas#16659]
xref #16639
```
In [25]: c1 = pd.Categorical(['a', 'b'])

In [26]: c2 = pd.Categorical(['a', 'b'])

In [27]: s = pd.Series(c1)

In [29]: c1.is_dtype_equal(c2)
Out[29]: True

In [30]: c1.is_dtype_equal(s)
Out[30]: False
```

I think `[30]` should be true?

**Comment by jreback** [comment]
yep [30] should be True.

**Comment by jschendel** [comment]
Looks like this is fixed in 0.22.0 (so really probably fixed in 0.21.x):
```python
In [2]: pd.__version__
Out[2]: '0.22.0'

In [3]: c1 = pd.Categorical(['a', 'b'])

In [4]: c2 = pd.Categorical(['a', 'b'])

In [5]: s = pd.Series(c1)

In [6]: c1.is_dtype_equal(c2)
Out[6]: True

In [7]: c1.is_dtype_equal(s)
Out[7]: True
```
Could still use a test to prevent regression though.

**Comment by jreback** [comment]
if u can put up a test would be great
